src/ContactHub.js

```diff
@@ -85,17 +75,24 @@ const cleanCustomer = (data: APICustomer): Customer => {
         updatedAt: s.updatedAt && new Date(s.updatedAt)
       }));
 
-    /* Strip nulls and empty arrays recursively from `base` */
+    /* Strip nulls and empty arrays recursively */
     const base = { ...data.base, jobs, likes, subscriptions };
-    customer.base = (compact(base): BaseProperties);
+    customer.base = (compact(base));
 
-    // check for data.base is useful only for avoid flow warning
+    // check for data.base is useful only to avoid a Flow warning
     if (data.base && data.base.dob) {
       customer.base.dob = new Date(data.base.dob);
     }
 
   }
 
+  if (data.consents) {
+    customer.consents = (compact(data.consents));
+    if (data.consents && data.consents.disclaimer && data.consents.disclaimer.date) {
```

Comment:
minor: you're already checking if `data.consents` is null in the external if.

Reply:
Flow is complaining if I remove it, because it doesn't trust `compact(data.consents)`.